packages/webapp/src/webview/ui/components/App/App.tsx

```diff
@@ -21,12 +22,46 @@ initIcons();
  */
 export function App(): ReactElement {
     const appState = useSelector<AppState, AppState>((state) => state);
+    const resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
```

Comment:
Hi @unseen1980, 
this implementation causes a memory leak, I assume every render of this component adds an additional `ResizeObserver` and observes. You can reproduce this by running the extension, do a search query, zoom all in and then all out. For me it get's very slow until the extension is stuck.
I think we should use react hook `useEffect()` in this case, I've prepared a proposal in PR: https://github.com/SAP/guided-answers-extension/pull/280. If you agree to the approach, feel free to merge it into this branch.

Reply:
Thanks, Klaus, yes that was an issue. I used your suggestion 👍